Islandora/jsonld
# Drop drupal 10.1 and add drupal 10.4 to the testing matrix.

**GitHub Issue**: https://github.com/Islandora/documentation/issues/2324

* Other Relevant Links (Google Groups discussion, related pull requests, 
Release pull requests, etc.)

# What does this Pull Request do?

Updates the testing matrix.
Also deletes the workflow file for the 2.x branch since it's no longer under active development (and the workflow file for the 2.x branch is in the 2.x branch, if necessary)

# What's new?
Like the others, we add drupal 10.4.x-dev and remove drupal 10.1 from the testing matrix.

This module has no obvious features that warrant limiting the drupal core compatibility. 

* Does this change require documentation to be updated? no
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository
(i.e. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# How should this be tested?
Tests should run, and pass.

A description of what steps someone could take to:
* Reproduce the problem you are fixing (if applicable)
* Test that the Pull Request does what is intended.
* Please be as detailed as possible.
* Good testing instructions help get your PR completed faster.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this 
PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers
